maro/rl/training/algorithms/sac.py

```diff
@@ -0,0 +1,236 @@
+from dataclasses import dataclass
+from typing import Callable, Dict, Optional, Tuple
+
+import torch
+
+from maro.rl.model import QNet
+from maro.rl.policy import ContinuousRLPolicy, RLPolicy
+from maro.rl.training import AbsTrainOps, RandomReplayMemory, remote, RemoteOps, SingleAgentTrainer, TrainerParams
+from maro.rl.utils import get_torch_device, ndarray_to_tensor, TransitionBatch
+from maro.utils import clone
+
+
+@dataclass
+class SoftActorCriticParams(TrainerParams):
+    get_q_critic_net_func: Callable[[], QNet] = None
+    update_target_every: int = 5
+    random_overwrite: bool = False
+    entropy_coef: float = 0.1
+    num_epochs: int = 1
+    n_start_train: int = 0
+    reward_discount: float = 0.9
+    q_value_loss_cls: Callable = None
+    soft_update_coef: float = 1.0
+
+    def __post_init__(self) -> None:
+        assert self.get_q_critic_net_func is not None
+
+    def extract_ops_params(self) -> Dict[str, object]:
+        return {
+            "get_q_critic_net_func": self.get_q_critic_net_func,
+            "entropy_coef": self.entropy_coef,
+            "reward_discount": self.reward_discount,
+            "q_value_loss_cls": self.q_value_loss_cls,
+            "soft_update_coef": self.soft_update_coef,
+        }
+
+
+class SoftActorCriticOps(AbsTrainOps):
+    def __init__(
+        self,
+        name: str,
+        policy_creator: Callable[[], RLPolicy],
+        get_q_critic_net_func: Callable[[], QNet],
+        parallelism: int = 1,
+        *,
```

Comment:
Can we add "**args" to the base class to fix the potential issue without `*`?

Reply:
Let's just rewrite this part in "normal" fashion.